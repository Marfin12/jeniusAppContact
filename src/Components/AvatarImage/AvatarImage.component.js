import * as React from 'react';
import { View, TouchableHighlight, Image } from 'react-native';

import styles from './AvatarImage.styles';
import config from './AvatarImage.config';

const _renderImage = (image) => (
  <TouchableHighlight style={styles.imageContainer}>
    <Image 
      source={image}
      style={styles.image}
    />
  </TouchableHighlight>
);

const _renderLetter = (text) => (
  <View style={styles.container}>
    <Text style={styles.text} {...platform.getAccessibilityLabel('Avatar')}>
      {string.getNameInitial(text)}
    </Text>
  </View>
);

const AvatarImage = (props) => {
  const { firstName, lastName, image } = props;
  const text = firstName + lastName;

  if (image === 'N/A') {
    return _renderLetter(text);
  }

  return _renderImage(image);
}

AvatarImage.displayName = config.displayName;
AvatarImage.defaultProps = config.defaultProps;

export default AvatarImage;
