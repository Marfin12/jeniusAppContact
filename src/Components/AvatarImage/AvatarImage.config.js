const defaultProps = {
  firstName: '',
  lastName: '',
  size: '',
  image: ''
};

const displayName = 'AvatarImage';

export default {
  defaultProps,
  displayName,
};
