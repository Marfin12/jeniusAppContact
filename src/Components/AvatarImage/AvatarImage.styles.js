const styles = {
  textContainer: (size) => ({
    width: size,
    height: size,
    borderRadius: size / 2,
    justifyContent: 'center',
    overflow: 'hidden',
    backgroundColor: '#f31222'
  }),
  text: (size) => ({
    color: '#ffffff',
    fontSize: size / 2,
    textAlign: 'center',
    letterSpacing: 0
  }),
  imageContainer: (size) => ({
    marginLeft: 8,
    height: size,
    width: size,
    borderRadius: size / 2,
    overflow: 'hidden'
  }),
  image: (size) => ({
    height: size,
    width: size,
    borderRadius: size / 2,
  }),
};

export default styles;
