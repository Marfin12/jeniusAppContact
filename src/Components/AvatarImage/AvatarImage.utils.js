export const removeSpecialChars = (string) => {
    if (!string) return string;
    return string.replace(/[^a-zA-Z0-9 ]/gi, '');
};

const concatInitials = (firstWord, secondWord) => {
    const firstLetter = firstWord.charAt(0);
    const secondLetter = secondWord.charAt(0);
    return (firstLetter + secondLetter).toUpperCase();
};
  
export const checkImageURL = async (image, setValidImage) => {
    try {
      const res = await fetch(image);
      if (res.status === 404) setValidImage('N/A');
      else setValidImage(image)
    } catch(err) {
      return setValidImage('N/A')
    }
}

export const getNameInitial = (text) => {
    if (!text) {
      return '';
    }
  
    const words = removeSpecialChars(text)
                    .split(' ')
                    .filter(string => string.length > 0);
    const lastIndex = words.length - 1;
    if (words.length === 0) return '';
    if (words.length === 1) return words[0].charAt(0);
  
    return concatInitials(words[0], words[lastIndex]);
};