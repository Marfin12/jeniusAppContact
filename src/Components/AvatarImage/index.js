export { default } from './AvatarImage.component';
