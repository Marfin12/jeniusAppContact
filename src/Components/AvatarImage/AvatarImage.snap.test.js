import React from 'react';
import renderer from 'react-test-renderer';

import AvatarImage from './AvatarImage.component';
import config from './AvatarImage.config';

jest.mock('react-redux', () => {
  const ActualReactRedux = jest.requireActual('react-redux');
  const lightMode = {
    theme: false,
  };
  const darkMode = {
    theme: true,
  };
  return {
    ...ActualReactRedux,
    useSelector: jest
      .fn()
      .mockImplementationOnce(() => lightMode)
      .mockImplementationOnce(() => darkMode),
  };
});
jest.mock('../../Components/TextField', () => 'TextField');
jest.useFakeTimers();

afterEach(() => {
  jest.resetAllMocks();
});

describe('Avatar Component Default Theme', () => {
  const ComponentWrapper = renderer
    .create(<AvatarImage {...config.defaultProps} />)
    .toJSON();

  test('renders avatar component light mode', () => {
    expect(ComponentWrapper).toMatchSnapshot();
  });

  const mockedProps = {
    ...config.defaultProps,
    image: 'https://thumbs.dreamstime.com/b/example-red-tag-example-red-square-price-tag-117502755.jpg'
  }
  const ComponentWrapperWithImageUrl = renderer
    .create(<AvatarImage {...mockedProps} />)
    .toJSON();

  test('renders avatar component image url mode', () => {
    expect(ComponentWrapperWithImageUrl).toMatchSnapshot();
  });
});

describe('Avatar Component Dark Theme', () => {
  const ComponentWrapper = renderer
    .create(<AvatarImage {...config.defaultProps} />)
    .toJSON();

  test('renders avatar component dark mode', () => {
    expect(ComponentWrapper).toMatchSnapshot();
  });
});
