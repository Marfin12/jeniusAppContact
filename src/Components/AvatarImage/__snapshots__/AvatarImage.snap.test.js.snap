// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar Component Dark Theme renders avatar component dark mode 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "borderRadius": 0,
      "height": "",
      "marginLeft": 8,
      "overflow": "hidden",
      "width": "",
    }
  }
>
  <Image
    source={
      Object {
        "uri": "",
      }
    }
    style={
      Object {
        "borderRadius": 0,
        "height": "",
        "width": "",
      }
    }
  />
</View>
`;

exports[`Avatar Component Default Theme renders avatar component image url mode 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#f31222",
      "borderRadius": 0,
      "height": "",
      "justifyContent": "center",
      "overflow": "hidden",
      "width": "",
    }
  }
>
  <Text
    style={
      Object {
        "color": "#ffffff",
        "fontSize": 0,
        "letterSpacing": 0,
        "textAlign": "center",
      }
    }
  >
    
  </Text>
</View>
`;

exports[`Avatar Component Default Theme renders avatar component light mode 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "borderRadius": 0,
      "height": "",
      "marginLeft": 8,
      "overflow": "hidden",
      "width": "",
    }
  }
>
  <Image
    source={
      Object {
        "uri": "",
      }
    }
    style={
      Object {
        "borderRadius": 0,
        "height": "",
        "width": "",
      }
    }
  />
</View>
`;
